pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -1014,6 +1014,7 @@ func TranslateRouteMatch(vs config.Config, in *networking.HTTPMatchRequest, useE
 		} else {
 			matcher := translateHeaderMatch(name, stringMatch)
 			matcher.InvertMatch = true
+			matcher.TreatMissingHeaderAsEmpty = true
```

Comment:
If we set this, does it mean headers unset will be treated as present?

Reply:
my understanding is yes, and it's value is empty.

Reply:
Then i think this is not correct when user have a VS with ` withoutHeaders: regex: *`

Reply:
I'll test it later.

Reply:
Do you mean `regex: .*` or `regex: .` ? It seems like `regex: *` is not a correct regex.

Reply:
why not? 


It seems the code checks *

func isCatchAllStringMatch(in *networking.StringMatch) bool {
	if in == nil || in.MatchType == nil {
		return true
	}

	catchall := false

	switch m := in.MatchType.(type) {
	case *networking.StringMatch_Regex:
		catchall = m.Regex == "*"
	}

	return catchall
}